### PR TITLE
Better diaporama endpoints for front office

### PR DESCRIPTION
**Pull request by air-dex**
- Better diaporama endpoints for using diaporamas in front office.
- Better JS management, now with `{javascripts}` Smarty tags

### CODE REVIEW

File: Controller/DiaporamaFrontController.php [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+use Thelia\Core\HttpFoundation\Response;
+
+class DiaporamaFrontController extends BaseFrontController
+{
+    public function getDiaporamaHtmlAction($shortcode)
+    {
+        $width = $this->getRequest()->query->get('width');
+        $height = $this->getRequest()->query->get('height');
+        $event = new DiaporamaEvent();
+        $event->__set('image_width', $width);
```
Comment (lunika) [inline Controller/DiaporamaFrontController.php:28]: `$event->image_width = $width;`

method `__set` is a magic method : http://php.net/manual/en/language.oop5.overloading.php#object.set

Comment (air-dex) [inline Controller/DiaporamaFrontController.php:28]: `$event->image_width = $width;` sucks for encapsulation. Is there a better way to do it?

Comment (lunika) [inline Controller/DiaporamaFrontController.php:28]: well, `__set` is not really better

Comment (air-dex) [inline Controller/DiaporamaFrontController.php:28]: You're right. Moreover `$event->image_width = $width;` is more understandable.

Comment (lunika) [inline Controller/DiaporamaFrontController.php:28]: Why not declaring those properties in your class and then create getters and setters methods ? What did you try to do ?

Comment (lunika) [inline Controller/DiaporamaFrontController.php:28]: Same in https://github.com/thelia-modules/Diaporamas/blob/master/Action/DiaporamaAction.php#L77 don't use the magic method. Directly `$event->image_width`

Comment (air-dex) [inline Controller/DiaporamaFrontController.php:28]: Same thing than here. I'm on it too.

Comment (air-dex) [inline Controller/DiaporamaFrontController.php:28]: https://github.com/thelia-modules/Diaporamas/pull/5 should solve it.

**Pull request merged**